redpanda/values.yaml

```diff
@@ -41,20 +41,16 @@ image:
 license_key: ""
 
 #
-# Authentication
+# Authentication & SASL configuration
 auth:
-  #
-  # SASL configuration
-  sasl:
-    enabled: false
-    # user list
-    # TODO create user at startup
-    users:
-      - name: admin
-        # Password for the user. This will be used to generate a secret
-        # password: password
-        # If password isn't given, then the secretName must point to an already existing secret
-        # secretName: adminPassword
+  enabled: false
+  # Set `bootstrap: {}` to enable sasl but not create the `RP_BOOTSTRAP_USER` envvar
+  bootstrap:
+    user: admin
+    password: ""
+    secretName: ""
+  additionalSuperUsers: []
```

Comment:
This could remain under `users` (or `superusers` with a helper that uses `users` for backward compatibility). The creation of the bootstrap secret could be done by taking the `| first` of the list.

How does this set the passwords for the additional superusers?

Reply:
I wanted to split it out for a scenario where someone didn't want to use the bootstrap env var in favor of some external process. They could set the bootstrap key to an empty object and use the additionalUsers key to mark the users as superusers in the config. My preference is to make the values more explicit rather than do list operations in helm but it's not a huge deal to revert the change.

I actually removed my additions for seeding passwords for all users as I wasn't sure if that was wanted

Reply:
Our desire is to make it as easy as possible for helm/kubernetes novices. The Helm chart should do things in a prescriptive way that helps with best practices. Adding superusers through values is simple and easy. As a list, it's intuitive. For the more advanced users, we want to add the ability to pull passwords from a secret+key that would need to be pre-populated.